docker/dockerfile-generator.sh

```diff
@@ -186,41 +205,79 @@ printPreamble() {
 # SPDX-License-Identifier: Apache-2.0
 # ********************************************************************************
 
-FROM ubuntu:18.04
+FROM $IMAGE
 
 LABEL maintainer=\"AdoptOpenJDK <adoption-discuss@openjdk.java.net>\"
   " >> "$DOCKERFILE_PATH"
 }
 
-# Put in apt packages required for building a JDK
-printAptPackages() {
+
+printAptPackagesBase() {
   if [ ${COMMENTS} == true ]; then
     echo "
-# Install required OS tools
+# Install required OS tools to setup environment as .deb via apt-get
 # dirmngr, gpg-agent & coreutils are all required for the apt-add repository command" >> "$DOCKERFILE_PATH"
   fi
+  echo " 
+RUN $UBUNTU_PREAMBLE
+  && apt-get install -qq -y --no-install-recommends \\
+    curl \\
+    git \\
+    unzip \\
+    wget \\
+    zip " >> "$DOCKERFILE_PATH"
+  echo "
+RUN rm -rf /var/lib/apt/lists/*" >> "$DOCKERFILE_PATH"
+}
 
+printDnfPackagesBase() {
+  if [ ${COMMENTS} == true ]; then
+    echo "
+# Install required OS tools to setup environment as rpms via dnf" >> "$DOCKERFILE_PATH"
+  fi
+  local skipGpg="" # it may bite from time to time
+  #local skipGpg="--nogpgcheck"
+  local erasing="--allowerasing"
+  if [ ${DNF_INSTALL} = yum ] ; then
+    erasing=""
+  fi
+  if echo "${IMAGE}" | grep -e "stream8" -e "centos:7" ; then
+    echo " 
+RUN cd /etc/yum.repos.d/ ; sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
+RUN cd /etc/yum.repos.d/ ; sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-* " >> "$DOCKERFILE_PATH"
```

Comment:
This is going to need a follow-on change as the non-x64 ones (aarch64 and ppc64le) are under vault.centos.org/altarch (This wasn't a problem for the CentOS6 ones we translated in the past as we only used that one on x64)

Reply:
Really, it works for8:

```
tester@fedora:~/temurin-build$  sh   makejdk-any-platform.sh   -D centos:stream8   --custom-cacerts  jdk21u
Starting makejdk-any-platform.sh to configure, build (Adoptium)OpenJDK binary
Parsing opt: -D
Possible opt arg: centos:stream8
base image will be set to: centos:stream8
Parsing opt: --custom-cacerts
Possible opt arg: jdk21u
Setting version based on forest_name=jdk21u
Default branch set to BUILD_CONFIG[BRANCH]=master
Using source repository https://github.com/openjdk/jdk21u
Working dir is ./build/
[debug] COPY_MACOSX_FREE_FONT_LIB_FOR_JDK_FLAG=false
[debug] COPY_MACOSX_FREE_FONT_LIB_FOR_JRE_FLAG=false
JDK Image folder name: jdk
JRE Image folder name: jre
Removing Dockerfile* from docker/jdk21/aarch64/centos
centos:stream8
centos:stream8
centos:stream8
Dockerfile created at docker/jdk21/aarch64/centos/Dockerfile
...
--> c9d0997cfc9c
STEP 18/42: RUN chmod 755 /home/tester/temurin-build/workspace/config
--> 0034495152dd
STEP 19/42: RUN chown -R build /home/tester/temurin-build/workspace/config
--> 2754a7505488
STEP 20/42: RUN cd /etc/yum.repos.d/ ; sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
--> ce5684c8a2eb
STEP 21/42: RUN cd /etc/yum.repos.d/ ; sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-* 
--> a6ed67c4dbcc
STEP 22/42: RUN dnf  -y update --allowerasing
CentOS Stream 8 - AppStream                      22 MB/s |  25 MB     00:01    
CentOS Stream 8 - BaseOS                         11 MB/s | 7.0 MB     00:00    
CentOS Stream 8 - Extras                         39 kB/s |  17 kB     00:00    
CentOS Stream 8 - Extras common packages         18 kB/s | 8.0 kB     00:00    
Dependencies resolved.
================================================================================
 Package          Architecture    Version                  Repository      Size
================================================================================
Upgrading:
 hostname         aarch64         3.20-7.el8.0.1           baseos          32 k

Transaction Summary
================================================================================
Upgrade  1 Package

Total download size: 32 k
Downloading Packages:
hostname-3.20-7.el8.0.1.aarch64.rpm              45 kB/s |  32 kB     00:00    
--------------------------------------------------------------------------------
Total                                            45 kB/s |  32 kB     00:00     
CentOS Stream 8 - BaseOS                        1.6 MB/s | 1.6 kB     00:00    
Importing GPG key 0x8483C65D:
 Userid     : "CentOS (CentOS Official Signing Key) <security@centos.org>"
...
  Verifying        : hostname-3.20-6.el8.aarch64                            2/2 

Upgraded:
  hostname-3.20-7.el8.0.1.aarch64                                               

Complete!
--> f57ef513cfff
STEP 23/42: RUN dnf  -y install --allowerasing     bzip2-libs     bzip2     curl     git     unzip     wget     zip 
Last metadata expiration check: 0:00:09 ago on Tue Jul  2 10:00:52 2024.
Package bzip2-libs-1.0.6-26.el8.aarch64 is already installed.
Package curl-7.61.1-34.el8.aarch64 is already installed.
Dependencies resolved.
=======================================================================================
 Package                   Arch     Version                            Repo        Size
=======================================================================================
Installing:
 bzip2                     aarch64  1.0.6-26.el8                       baseos      60 k
 git                       aarch64  2.43.0-1.el8                       appstream   92 k
 unzip                     aarch64  6.0-46.el8                         baseos     189 k
 wget                      aarch64  1.19.5-11.el8                      appstream  717 k
 zip                       aarch64  3.0-23.el8                         baseos     265 k
Installing dependencies:
...
```

Reply:
The alt-arch was only cento7 speciality. So I guess you were referring to it. Sorry, I misread a bit. I will try alsocentos7 on aarch64, and I guess thatr will need the alt for sure.

Reply:
Again, my solution is pretty similar.. wrapped in an if statement..  
  
`sed -i 's|#baseurl=http://mirror.centos.org/altarch/\$releasever/|baseurl=http://vault.centos.org/altarch/7.9.2009/|' /etc/yum.repos.d/CentOS-Base.repo; \`  
  
\`\`\`  
RUN sed -i -e 's!mirrorlist!#mirrorlist!g' /etc/yum.repos.d/CentOS-Base.repo

RUN if \[ "$(uname -m)" = "aarch64" \]; then \\  
     echo "Running on arm64 architecture"; \\  
     sed -i 's|#baseurl=http://mirror.centos.org/altarch/\$releasever/|baseurl=http://vault.centos.org/altarch/7.9.2009/|' /etc/yum.repos.d/CentOS-Base.repo; \\  
   elif \[ "$(uname -m)" = "ppc64le" \]; then \\  
     echo "Running on ppc64le architecture"; \\  
     sed -i 's|#baseurl=http://mirror.centos.org/altarch/\$releasever/|baseurl=http://vault.centos.org/altarch/7.9.2009/|' /etc/yum.repos.d/CentOS-Base.repo; \\  
   else \\  
     # Default To x64; \\  
     echo "Running on another architecture"; \\  
     sed -i -e 's!#baseurl=http://mirror.centos.org/centos/\$releasever!baseurl=https://vault.centos.org/7.9.2009/!g' /etc/yum.repos.d/CentOS-Base.repo; \\  
   fi  
\`\`\`

Reply:
> Again, my solution is pretty similar.. wrapped in an if statement..
> 
> `sed -i 's|#baseurl=http://mirror.centos.org/altarch/\$releasever/|baseurl=http://vault.centos.org/altarch/7.9.2009/|' /etc/yum.repos.d/CentOS-Base.repo; \`
> 
> ```
> RUN sed -i -e 's!mirrorlist!#mirrorlist!g' /etc/yum.repos.d/CentOS-Base.repo
> 
> RUN if \[ "$(uname -m)" = "aarch64" \]; then \\  
>      echo "Running on arm64 architecture"; \\  
>      sed -i 's|#baseurl=http://mirror.centos.org/altarch/\$releasever/|baseurl=http://vault.centos.org/altarch/7.9.2009/|' /etc/yum.repos.d/CentOS-Base.repo; \\  
>    elif \[ "$(uname -m)" = "ppc64le" \]; then \\  
>      echo "Running on ppc64le architecture"; \\  
>      sed -i 's|#baseurl=http://mirror.centos.org/altarch/\$releasever/|baseurl=http://vault.centos.org/altarch/7.9.2009/|' /etc/yum.repos.d/CentOS-Base.repo; \\  
>    else \\  
>      # Default To x64; \\  
>      echo "Running on another architecture"; \\  
>      sed -i -e 's!#baseurl=http://mirror.centos.org/centos/\$releasever!baseurl=https://vault.centos.org/7.9.2009/!g' /etc/yum.repos.d/CentOS-Base.repo; \\  
>    fi  
> ```

Thanx!! I will reuse it immediately if it bites back or if @sxa insits.